### DISCUSSION TITLE
WalletConnectSignClient.Connect() method not working in Unity Editor

### DISCUSSION THREAD

**Discussion by yozdev** [WalletConnect/walletconnect-monorepo#3216] [Developer Support]
Hello Team,

This is with reference to the latest WalletConnectUnity project pulled from https://github.com/WalletConnect/WalletConnectUnity v2.0 branch.

I'm using Unity Editor version 2021.3.23f1 and building on Windows platform

After calling the Connect() method of WCSignClient.cs, WalletConnectSignClient.Connect() method is not getting called.

Following is my code :

1) I call WCSignClient.Connect() as shown below

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/113970224/51efacd9-5b1a-42b5-9a8a-8f1138a2408d)

2) Following is the code in WCSignClient.cs

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/113970224/cb610039-b10a-407b-a25a-860de6a44bea)

I get the console outputs for 

Debug.Log("WCSignClient Connect called");

foreach (var key in options.RequiredNamespaces.Keys)
{
        Debug.Log($"Key: {key}, Value: {options.RequiredNamespaces[key].Chains[0]}");
}

Following the snap of the console output I get in the editor

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/113970224/fb49d596-8ba1-4ea7-854a-634ed98b5359)

However, code is not executed beyond this.

Please help me resolve this.

**Comment by mustafadervisglu** [comment]
did you solve this error? @yozdev

**Reply by yozdev** [reply]
Hey @mustafadervisglu

I haven't found a solution yet

**Reply by mustafadervisglu** [reply]
We use it in our game, and currently, the inability to use the wallet has caused significant inconvenience for us in the game. @yozdev

**Reply by yozdev** [reply]
@mustafadervisglu we are facing the same problem

**Comment by yozdev** [comment]
This has been resolved. Thank you team.

**Reply by mustafadervisglu** [reply]
how it was solved @yozdev

**Reply by yozdev** [reply]
@mustafadervisglu it's working in the builds now after the latest git push. Please pull the latest code and check again.
